consoles/vnc_base.pm

```diff
@@ -191,31 +168,17 @@ sub mouse_hide {
     return {absolute => $self->{vnc}->absolute};
 }
 
-sub mouse_set {
-    my ($self, $args) = @_;
+sub mouse_set ($self, $args) {
     die "Need x/y arguments" unless (defined $args->{x} && defined $args->{y});
-
-    # TODO: for framebuffers larger than 1024x768, we need to upscale
     $self->_mouse_move(int($args->{x}), int($args->{y}));
     return {};
 }
 
-sub mouse_button {
-    my ($self, $args) = @_;
-
+sub mouse_button ($self, $args) {
     my $button = $args->{button};
     my $bstate = $args->{bstate};
 
-    my $mask = 0;
-    if ($button eq 'left') {
-        $mask = $bstate;
-    }
-    elsif ($button eq 'right') {
-        $mask = $bstate << 2;
-    }
-    elsif ($button eq 'middle') {
-        $mask = $bstate << 1;
-    }
+    my $mask = $button eq 'left' ? $bstate : $button eq 'right' ? $bstate << 2 : $bstate << 1;
```

Comment:
Pressing the middle button when the specified button is actually invalid seems a bit random. Previously the `$mask` was zero in that case and I'd rather preserve that behavior and possibly print a warning or even raise an error in that case. Besides, in the previous code `1` wasn't a magic number because it was clear that it mean "middle". In the changed version that's no longer clear.

Reply:
yes, you are right. Updated with a nicer lookup-table approach.

Reply:
Looks good :-)